lib/site_setting_extension.rb

```diff
@@ -544,11 +543,11 @@ def setup_methods(name)
       end
     else
       define_singleton_method clean_name do
-        if (c = current[name]).nil?
+        if (current_value = current[name]).nil?
           refresh!
           current[name]
         else
-          c
```

Comment:
This is an extremely hot path - I don't think we should introduce a modifier here. Even without any plugin overrides, it will have some performance impact on every single site setting call. 

Instead, I think we should apply the modifier when building the cache. e.g. for `site_setting_defaults` is applied here:

https://github.com/discourse/discourse/blob/83f1a13374bc3906919744f48071511c12912df9/lib/site_settings/defaults_provider.rb#L41-L41

This seems like a good place to add a a modifier which allows overriding DB values:

https://github.com/discourse/discourse/blob/83f1a13374bc3906919744f48071511c12912df9/lib/site_setting_extension.rb#L281-L292

Couple other thoughts:

1. Should this override env-based overrides? I think not - env should always win
2. Can we call it more specific like `site_setting_values` (to distinguish from the existing `site_setting_defaults`